src/substrate/filters/enricher.ts

```diff
@@ -66,6 +67,25 @@ export async function Enrich(
         }
       }
 
+      /**
+       * Offences Events
+       */
+      case EventKind.Offence: {
+        const [ offenceKind, opaqueTimeSlot, applied ] = event.data as unknown as [ Kind, OpaqueTimeSlot, boolean ];
+        const reportIds = await api.query.offences.concurrentReportsIndex(offenceKind, opaqueTimeSlot);
+        const offenceDetails: Option<OffenceDetails>[] = await api.query.offences.reports
+          .multi(reportIds.map((reportId) => reportId.toString()));
+        const offenders : Array<ValidatorId> = offenceDetails.map((offence) => offence.unwrap().offender[0]);
+        return {
+          data: {
+            kind,
+            offenceKind: offenceKind.toString(),
+            opaqueTimeSlot: opaqueTimeSlot.toString(),
+            applied: typeof applied === 'undefined' ? true : applied,
```

Comment:
This conditional should be easier to manage with the `bool` change I suggested above.